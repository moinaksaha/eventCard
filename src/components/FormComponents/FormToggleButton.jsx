import React, { Component } from 'react';

// import { FormGroup, Radio, ControlLabel } from 'react-bootstrap'

import styles from '../../containers/Home/Home.css';

export default class FormToggleButton extends Component{

	constructor(){
    	super();
	}
	  
	handleChange = (event) => {
		const { togglevalue, setToggleValue } = this.props;
		setToggleValue(togglevalue);
	}

	render = () => {

        const { toggleID, toggleName, togglevalue, labelText } = this.props;
		
		return (

			<div>

				<label className={`${styles.switch}`} 
					   htmlFor={toggleID}> 
                  
            		<div className={`${styles.customLabelText}`}>{labelText}</div> 

					<input type="radio" 
						   id={toggleID} 
						   name={toggleName} 
						   value={togglevalue} 
						   onChange={this.handleChange}
						   ref="radioButton"/>

                	<div className={`${styles.slider} ${styles.round}`}></div>

                </label>

            </div>

		);

	}
	
};

FormToggleButton.defaultProps = {

};
