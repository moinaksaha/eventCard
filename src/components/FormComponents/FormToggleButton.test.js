import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

import FormToggleButton from './FormToggleButton';

const mockRemoveGreeting = jest.fn();

// describe what we are testing
describe('Form Toggle Component', () => {
 
    it('renders correctly', () => {
        const successSnap = renderer.create(
          <FormToggleButton />
        ).toJSON();
        expect(successSnap).toMatchSnapshot();
    });

})


describe('Radio Toggle input', () => {
  
    it('should respond to change event and change the state of the Login Component', () => {
        
        const wrapper = shallow(<FormToggleButton setToggleValue={mockRemoveGreeting}/>);
        wrapper.find('input').simulate('change', { target: { checked: true } });
        expect(mockRemoveGreeting).toBeCalled();
    })
})