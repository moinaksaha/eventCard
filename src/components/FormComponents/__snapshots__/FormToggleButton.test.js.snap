// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Toggle Component renders correctly 1`] = `
<div>
  <label
    className="undefined"
    htmlFor={undefined}
  >
    <div
      className="undefined"
    />
    <input
      id={undefined}
      name={undefined}
      onChange={[Function]}
      type="radio"
      value={undefined}
    />
    <div
      className="undefined undefined"
    />
  </label>
</div>
`;
